File: C#.Net/Tests/JDI UIWebTests/JDI UIWebTests/Tests/Complex/ComboBoxTests.cs
using System;
using NUnit.Framework;
using System.Collections.Generic;
using Epam.JDI.Core.Interfaces.Complex;
using static Epam.JDI.Core.Settings.JDISettings;
using static JDI_UIWebTests.UIObjects.TestSite;
using static JDI_UIWebTests.Enums.Metals;
using JDI_UIWebTests.Enums;
using JDI_Matchers.NUnit;
using static JDI_UIWebTests.Tests.Complex.CommonActionsData;
using JDI_Web.Settings;
using OpenQA.Selenium;
using Assert = JDI_Matchers.NUnit.Assert;

namespace JDI_UIWebTests.Tests.Complex
{
    public class ComboBoxTests
    {
        private static readonly IList<string> OddOptions = new List<string> { "Col", "Gold", "Silver", "Bronze", "Selen" };

        private IComboBox<Metals> MetalsControl => MetalsColorsPage.ComboBox;

        [SetUp]
        public void SetUp()
        {
            Logger.Info("Navigating to Metals and Colors page.");
            MetalsColorsPage.Open();
            MetalsColorsPage.CheckTitle();
            MetalsColorsPage.IsOpened();
            Logger.Info("Setup method finished");
            Logger.Info("Start test: " + TestContext.CurrentContext.Test.Name);
        }

        [Test]
        public void SelectStringTest()
        {
            MetalsControl.Select("Gold");
            CheckAction("Metals: value changed to Gold");
        }

        [Test]
        public void SelectIndexTest()
        {
            MetalsControl.Select(3);
            CheckAction("Metals: value changed to Silver");
        }

        [Test]
        public void SelectEnumTest()
        {
            MetalsControl.Select(Gold);
            CheckAction("Metals: value changed to Gold");
        }

        [Test]
        public void GetOptionsTest()
        {
            new Check().CollectionEquals(MetalsControl.Options, OddOptions);
        }

        [Test]
        public void GetNamesTest()
        {
            new Check().CollectionEquals(MetalsControl.Names, OddOptions);
        }

        [Test]
        public void GetValuesTest()
        {
            new Check().CollectionEquals(MetalsControl.Values, OddOptions);
        }

        [Test]
        public void GetOptionsAsTextTest()
        {
            new Check().AreEquals(MetalsControl.OptionsAsText, "Col, Gold, Silver, Bronze, Selen");
        }

        [Test]
        public void SetValueTest()
        {
            MetalsControl.Value = "Blue";
            WebSettings.WebDriver.FindElement(By.ClassName("footer-content")).Click();
            CheckAction("Metals: value changed to Blue");
        }

        [Test]
        public void GetSelectedTest()
        {
            MetalsControl.Select("Gold");
            new Check().AreEquals(MetalsControl.Selected(), "Gold");
        }

        [Test]
        public void GetSelectedIndexTest()
        {
            CheckActionThrowError(() => MetalsControl.SelectedIndex(), NoElementsMessage);
        }

        [Test]
        public void IsSelectedTest()
        {
            Assert.AreEquals(MetalsControl.Selected("Col"), true);
        }

        [Test]
        public void IsSelectedEnumTest()
        {
            Assert.AreEquals(MetalsControl.Selected(Col), true);
        }

        [Test]
        public void WaitSelectedTest()
        {
            try
            {
                MetalsControl.WaitSelected("Col");
            }
            catch (Exception ex)
            {
                throw Exception("WaitSelected throws exception");
            }
        }

        [Test]
        public void WaitSelectedEnumTest()
        {
            new Check("WaitSelected").HasNoException(() => MetalsControl.Value.ToString());
        }
         
        [Test]
        public void GetValueTest()
        {
            Assert.AreEquals(MetalsControl.Value, "Col");
        }        
    }
}

File: Java/Tests/jdi-uitest-webtests/src/test/java/com/epam/jdi/uitests/testing/unittests/tests/complex/ComboBoxTests.java
package com.epam.jdi.uitests.testing.unittests.tests.complex;

import com.epam.jdi.uitests.core.interfaces.complex.IComboBox;
import com.epam.jdi.uitests.testing.unittests.InitTests;
import com.epam.jdi.uitests.testing.unittests.enums.Metals;
import com.epam.web.matcher.testng.Check;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Test;

import java.lang.reflect.Method;
import java.util.List;

import static com.epam.jdi.uitests.core.preconditions.PreconditionsState.isInState;
import static com.epam.jdi.uitests.testing.unittests.enums.Metals.Metals;
import static com.epam.jdi.uitests.testing.unittests.enums.Metals.Gold;
import static com.epam.jdi.uitests.testing.unittests.enums.Preconditions.METALS_AND_COLORS_PAGE;
import static com.epam.jdi.uitests.testing.unittests.pageobjects.EpamJDISite.metalsColorsPage;
import static com.epam.jdi.uitests.testing.unittests.tests.complex.CommonActionsData.*;
import static com.epam.web.matcher.testng.Assert.*;
import static java.util.Arrays.asList;

/**
 * Created by Roman_Iovlev on 9/15/2015.
 */
public class ComboBoxTests extends InitTests {
    private static final List<String> oddOptions = asList("Metals", "Gold", "Silver", "Bronze", "Selen");

    private IComboBox<Metals> metals() {
        return metalsColorsPage.comboBox;
    }

    @BeforeMethod
    public void before(Method method) {
        isInState(METALS_AND_COLORS_PAGE, method);
    }

    @Test
    public void selectStringTest() {
        metals().select("Gold");
        checkAction("Metals: value changed to Gold");
    }

    @Test
    public void selectIndexTest() {
        metals().select(3);
        checkAction("Metals: value changed to Silver");
    }

    @Test
    public void selectEnumTest() {
        metals().select(Gold);
        checkAction("Metals: value changed to Gold");
    }

    @Test
    public void getOptionsTest() {
        listEquals(metals().getOptions(), oddOptions);
    }

    @Test
    public void getNamesTest() {
        listEquals(metals().getNames(), oddOptions);
    }

    @Test
    public void getValuesTest() {
        listEquals(metals().getValues(), oddOptions);
    }

    @Test
    public void getOptionsAsTextTest() {
        areEquals(metals().getOptionsAsText(), "Metals, Gold, Silver, Bronze, Selen");
    }

    @Test
    public void setValueTest() {
        metals().setValue("Blue");
        looseFocus();
        checkAction("Metals: value changed to Blue");
    }

    @Test
    public void getSelectedTest() {
        metals().select("Gold");
        areEquals(metals().getSelected(), "Gold");
    }

    @Test
    public void getSelectedIndexTest() {
        checkActionThrowError(() -> metals().getSelectedIndex(), noElementsMessage); // isDisplayed not defined
    }

    @Test
    public void isSelectedTest() {
        areEquals(metals().isSelected("Metals"), true);
    }

    @Test
    public void isSelectedEnumTest() {
        areEquals(metals().isSelected(Metals), true);
    }

    @Test
    public void waitSelectedTest() {
        try {
            metals().waitSelected("Metal");
        } catch (Exception ex) {
            throw exception("WaitSelected throws exception");
        }
    }

    @Test
    public void waitSelectedEnumTest() {
        new Check("WaitSelected")
            .hasNoExceptions(() -> metals().waitSelected(Metals));
    }

    @Test
    public void getValueTest() {
        areEquals(metals().getValue(), "Metals");
    }
}
